bin/compile

```diff
@@ -1,91 +1,76 @@
 ORACLE_INSTANT_CLIENT_URL="http://buildpacks.teresa-apps.io/oracle/oracle-instantclient-full-12.1-linux.x64.tar.gz"
-ORACLE_INSTANT_CLIENT_FILENAME=ORACLE_INSTANT_CLIENT_URL.split(/\//).last
+ORACLE_INSTANT_CLIENT_FILENAME=$(echo $ORACLE_INSTANT_CLIENT_URL | cut -d/ -f5)
 
-indent "Making directory #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}"
-`mkdir -p #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}`
+source "$BIN_DIR/utils"
 
-indent "Downloading and extracting #{ORACLE_INSTANT_CLIENT_FILENAME}"
-result = `curl #{ORACLE_INSTANT_CLIENT_URL} -L -s -o - | tar -xvz -C #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR} -f - `
+puts-step "Found an .oracle.yml"
 
-unless $?.success?
-  indent "Failure while downloading Oracle instant client archive: #{$?}"
+indent "Making directory ${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR}"
+mkdir -p ${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR}
+
+indent "Downloading and extracting ${ORACLE_INSTANT_CLIENT_FILENAME}"
+curl ${ORACLE_INSTANT_CLIENT_URL} -L -s -o - | tar -xz -C \
+	${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR} -f -
+
+if [[ $? != 0 ]]; then
+  indent "Failure while downloading Oracle instant client archive: $?"
   exit 1
-end
+fi
 
-indent "Successfully extracted #{ORACLE_INSTANT_CLIENT_FILENAME}"
+indent "Successfully extracted ${ORACLE_INSTANT_CLIENT_FILENAME}"
 
-ORACLE_NETWORK_ADMIN_DIR = "#{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}/network/admin"
-`mkdir -p #{ORACLE_NETWORK_ADMIN_DIR}`
+ORACLE_NETWORK_ADMIN_DIR="${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR}/network/admin"
+mkdir -p ${ORACLE_NETWORK_ADMIN_DIR}
 
-if ORACLE_CONFIG
-  TNSNAMES_FILENAME = ORACLE_CONFIG['tnsnames.ora']
-  if File.exists?(File.join(BUILD_DIR,TNSNAMES_FILENAME))
-    indent "Symlinking #{TNSNAMES_FILENAME} to #{ORACLE_INSTANT_CLIENT_DIR}/network/admin/tnsnames.ora"
-    `ln -sf #{File.join('/app',TNSNAMES_FILENAME)} #{ORACLE_NETWORK_ADMIN_DIR}/tnsnames.ora`
-  end
+if [[ -s $ORACLE_CONFIG ]]; then
+  TNSNAMES_FILENAME=$(cat $ORACLE_CONFIG | grep tnsnames.ora | cut -d' ' -f2)
+  if [[ -f $BUILD_DIR/$TNSNAMES_FILENAME ]]; then
+    indent "Symlinking ${TNSNAMES_FILENAME} to ${ORACLE_INSTANT_CLIENT_DIR}/network/admin/tnsnames.ora"
+    ln -sf $BUILD_DIR/$TNSNAMES_FILENAME ${ORACLE_NETWORK_ADMIN_DIR}/tnsnames.ora
+  fi
 
-  SQLNET_FILENAME = ORACLE_CONFIG['sqlnet.ora']
-  if File.exists?(File.join(BUILD_DIR,SQLNET_FILENAME))
-    indent "Symlinking #{SQLNET_FILENAME} to #{ORACLE_INSTANT_CLIENT_DIR}/network/admin/sqlnet.ora"
-    `ln -sf #{File.join('/app',SQLNET_FILENAME)} #{ORACLE_NETWORK_ADMIN_DIR}/sqlnet.ora`
-  end
+  SQLNET_FILENAME=$(cat $ORACLE_CONFIG | grep sqlnet.ora | cut -d' ' -f2)
+  if [[ -f $BUILD_DIR/$SQLNET_FILENAME ]]; then
+    indent "Symlinking ${SQLNET_FILENAME} to ${ORACLE_INSTANT_CLIENT_DIR}/network/admin/sqlnet.ora"
+    ln -sf $BUILD_DIR/$SQLNET_FILENAME ${ORACLE_NETWORK_ADMIN_DIR}/sqlnet.ora
+  fi
 else
-  warning '.oracle.yml is empty or invalid'
-end
+  puts-warn ".oracle.yml is empty or invalid"
```

Comment:
What do you think about a more friendly message, like "There is no sqlnet.ora entry on .oracle.yml" ? (just to clarify WTF?!? is wrong with .oracle.yml)

Reply:
Make sense, maybe `.oracle.yml is empty`? This is the only case tested

Reply:
BTW I'm not sure if it this warning is needed, the buildpack will work fine fine with a empty file.

Reply:
🤔 IMO it's ok to remove warn

Reply:
removed